pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -52,7 +52,8 @@ func (e *ErrorWithCodes) Error() string {
 }
 
 var (
-	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|InvalidAuthenticationTokenTenant|SignatureDoesNotMatch|Authentication failed|AuthFailure|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId|AccessDenied|OperationNotAllowed|Error 403)`)
+	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|SignatureDoesNotMatch|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId|AccessDenied|OperationNotAllowed|Error 403)`)
+	unauthenticatedRegexp               = regexp.MustCompile(`(?i)(InvalidAuthenticationTokenTenant|Authentication failed|AuthFailure)`)
```

Comment:
did I pick the right errors? should I move any others from unauthorized?

Reply:
I guess we should probably add InvalidAccessKeyId|InvalidSecretAccessKey|InvalidClientTokenId|InvalidSubscriptionId to Unauthenticated error code. WDYT?

Reply:
Most of them look fine. Please google the error strings you are unsure of and check for example here:
https://docs.aws.amazon.com/STS/latest/APIReference/CommonErrors.html
See also https://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses